### DISCUSSION TITLE
Are the warnings occurring with the "colour.plotting.plot_chromaticity_diagram_CIE1976UCS" definition an issue?

### DISCUSSION THREAD

**Discussion by blue-j** [colour-science/colour#1086] [Q&A] [answered]
I have tried for hours, and just can't crack it!  Does anyone know what is wrong with this code?

`import colour
import numpy as np

XYZ = np.array([0.95045, 1, 1.08905])
LUV = colour.XYZ_to_Luv(XYZ)
uv = colour.Luv_to_uv(LUV)
u = uv[0]
v = uv[1]

figure, axes = colour.plotting.plot_chromaticity_diagram_CIE1976UCS(
    show_spectral_locus=True,
    show_diagram_colours=True,
    diagram_opacity=1,
    title=None,
    standalone=False,
)

axes.plot(u, v, "o-", markersize=10, color="black")

colour.plotting.render(
    x_ticker=True,
    y_ticker=True,
    x_label=None,
    y_label=None,
    axes_visible=True,
    background_transparency=True,
    filename="custom_diagram.svg"
)`

I get

/colour/models/cie_luv.py:386: RuntimeWarning: divide by zero encountered in divide
  xy = tstack([9 * u / d, 4 * v / d])
/colour/models/cie_xyy.py:179: RuntimeWarning: invalid value encountered in multiply
  [x[m_XYZ] * Y_y, Y[m_XYZ], (1 - x[m_XYZ] - y[m_XYZ]) * Y_y]

Thanks in advance for any help you might offer.  : )

- J

**Comment by thomasmansencal** [comment]
Hey @blue-j,

What issue do you have? Seems to work just fine:

![image](https://user-images.githubusercontent.com/99779/207778435-55f11173-6c7d-42cb-a983-86d1c799e69f.png)

Cheers,

Thomas

**Comment by fredsavoir** [comment]
Hi,

Same issues on my hardware.
divide by zero encountered in divide
  xy = tstack([9 * u / d, 4 * v / d])
 RuntimeWarning: invalid value encountered in multiply
  [x[m_XY

Interesting.

Fred

**Reply by blue-j** [reply]
I'm unfortunately going to bed but this may be a dependency issue?  I'm using the latest colour but maybe we ought to compare our versions of numpy? I did not get a chance to look at the code throwing the error either... - J

**Comment by fredsavoir** [comment]
Hi,

I did tests with Python3.10.8 @ Python3.11.0 = same issues.

Name: colour-science
Version: 0.4.2

Name: numpy
Version: 1.23.5

Name: scipy
Version: 1.9.3

Name: imageio
Version: 2.22.4

Name: matplotlib
Version: 3.6.2

Name: typing_extensions
Version: 4.4.0

Same for you?
Thanks

Fred

**Comment by fredsavoir** [comment]
And the latest develop do the same on me.

**Comment by thomasmansencal** [comment, marked as answer]
The warnings are not an issue, they occur when the chromaticity diagram colours are being computed, I thought I removed those but apparently not so I did another pass. There are a lot of divisions in Colour and we decided to handle the obvious ones that might get to 0 in the denominator. They should have been tamed in develop.

**Reply by blue-j** [reply]
It's not only warnings for me. The plot of the point never occurs. - J

**Reply by thomasmansencal** [reply]
Ah! It is because you pass the `filename` argument so depending on the backend the figure might not be shown at all! Try commenting the line.

**Reply by blue-j** [reply]
Woot!

Well, i see the plot of the value now!

this is out:

```
/usr/local/lib/python3.10/site-packages/colour/utilities/verbose.py:266: ColourUsageWarning: The "CIE 2012 2 Degree Standard Observer" has been renamed to "CIE 2015 2 Degree Standard Observer" for consistency with the official CIE name which was adopted in 2015.
  warn(*args, **kwargs)
/usr/local/lib/python3.10/site-packages/colour/utilities/verbose.py:266: ColourUsageWarning: The "CIE 2012 10 Degree Standard Observer" has been renamed to "CIE 2015 10 Degree Standard Observer" for consistency with the official CIE name which was adopted in 2015.
  warn(*args, **kwargs)
/usr/local/lib/python3.10/site-packages/colour/models/cie_luv.py:386: RuntimeWarning: divide by zero encountered in divide
  xy = tstack([9 * u / d, 4 * v / d])
/usr/local/lib/python3.10/site-packages/colour/models/cie_xyy.py:179: RuntimeWarning: invalid value encountered in multiply
  [x[m_XYZ] * Y_y, Y[m_XYZ], (1 - x[m_XYZ] - y[m_XYZ]) * Y_y]
```

resulting from:

`import colour
import numpy as np

XYZ = np.array([0.95045, 1, 1.08905])
LUV = colour.XYZ_to_Luv(XYZ)
uv = colour.Luv_to_uv(LUV)
u = uv[0]
v = uv[1]

figure, axes = colour.plotting.plot_chromaticity_diagram_CIE1976UCS(
    show_spectral_locus=True,
    show_diagram_colours=True,
    diagram_opacity=1,
    title=None,
    standalone=False,
)

axes.plot(u, v, "o-", markersize=10, color="black")

colour.plotting.render(
    x_ticker=True,
    y_ticker=True,
    x_label=None,
    y_label=None,
    axes_visible=True,
    background_transparency=True,
)
`
this line commented out:   `filename="custom_diagram.svg"`


![Screen Shot 2022-12-15 at 11 56 36 AM](https://user-images.githubusercontent.com/3990142/207954841-92a8de40-d9e5-4a17-a34d-8749a433ec96.png)

**Comment by fredsavoir** [comment]
Me it’s displaying.

Let’s check your version of python & package.

Fred

**Comment by fredsavoir** [comment]
Check with pipdate to make sure all your package are up-to-date :)

**Comment by fredsavoir** [comment]
pip3 list ==

Package            Version
------------------ ---------
appdirs            1.4.4
certifi            2022.12.7
charset-normalizer 3.0.1
colour-science     0.4.2
commonmark         0.9.1
contourpy          1.0.6
cycler             0.11.0
fonttools          4.38.0
idna               3.4
imageio            2.22.4
kiwisolver         1.4.4
matplotlib         3.6.2
numpy              1.23.5
packaging          22.0
Pillow             9.3.0
pip                22.3.1
pipdate            0.5.6
Pygments           2.13.0
pyparsing          3.0.9
python-dateutil    2.8.2
requests           2.28.1
rich               12.6.0
scipy              1.9.3
setuptools         65.6.3
six                1.16.0
typing_extensions  4.4.0
urllib3            1.26.13
wheel              0.38.4

On my machine and you?
Fred

**Reply by blue-j** [reply]
Package                                           Version
------------------------------------------------- ------------
autocommand                                       2.2.2
boto3                                             1.26.27
botocore                                          1.29.27
cachetools                                        5.2.0
certifi                                           2022.12.7
charset-normalizer                                2.1.1
cheroot                                           9.0.0
CherryPy                                          18.8.0
click                                             8.1.3
colormath                                         3.0.0
colour-science                                    0.4.2
contourpy                                         1.0.6
cycler                                            0.11.0
et-xmlfile                                        1.1.0
fonttools                                         4.38.0
future                                            0.18.2
google-api-core                                   2.11.0
google-auth                                       2.15.0
google-cloud-texttospeech                         2.12.3
googleapis-common-protos                          1.57.0
graphviz                                          0.20.1
grpcio                                            1.51.1
grpcio-status                                     1.51.1
gTTS                                              2.3.0
idna                                              3.4
imageio                                           2.22.4
inflect                                           6.0.2
jaraco.classes                                    3.2.3
jaraco.collections                                3.8.0
jaraco.context                                    4.2.0
jaraco.functools                                  3.5.2
jaraco.text                                       3.11.0
jmespath                                          1.0.1
kiwisolver                                        1.4.4
latest                                            0.6.0
lensfun                                           0.3.3
matplotlib                                        3.6.2
more-itertools                                    9.0.0
networkx                                          2.8.8
numpy                                             1.23.5
openai                                            0.25.0
opencolorio                                       2.2.0
openpyxl                                          3.0.10
packaging                                         22.0
pandas                                            1.5.2
pandas-stubs                                      1.5.2.221124
Pillow                                            9.3.0
pip                                               22.3.1
portend                                           3.1.0
proto-plus                                        1.22.1
protobuf                                          4.21.11
pyasn1                                            0.4.8
pyasn1-modules                                    0.2.8
PyAudio                                           0.2.12
pybind11                                          2.10.1
pydantic                                          1.10.2
pyobjc                                            9.0
pyobjc-core                                       9.0
pyobjc-framework-Accessibility                    9.0
pyobjc-framework-Accounts                         9.0
pyobjc-framework-AddressBook                      9.0
pyobjc-framework-AdServices                       9.0
pyobjc-framework-AdSupport                        9.0
pyobjc-framework-AppleScriptKit                   9.0
pyobjc-framework-AppleScriptObjC                  9.0
pyobjc-framework-ApplicationServices              9.0
pyobjc-framework-AppTrackingTransparency          9.0
pyobjc-framework-AudioVideoBridging               9.0
pyobjc-framework-AuthenticationServices           9.0
pyobjc-framework-AutomaticAssessmentConfiguration 9.0
pyobjc-framework-Automator                        9.0
pyobjc-framework-AVFoundation                     9.0
pyobjc-framework-AVKit                            9.0
pyobjc-framework-BusinessChat                     9.0
pyobjc-framework-CalendarStore                    9.0
pyobjc-framework-CallKit                          9.0
pyobjc-framework-CFNetwork                        9.0
pyobjc-framework-ClassKit                         9.0
pyobjc-framework-CloudKit                         9.0
pyobjc-framework-Cocoa                            9.0
pyobjc-framework-Collaboration                    9.0
pyobjc-framework-ColorSync                        9.0
pyobjc-framework-Contacts                         9.0
pyobjc-framework-ContactsUI                       9.0
pyobjc-framework-CoreAudio                        9.0
pyobjc-framework-CoreAudioKit                     9.0
pyobjc-framework-CoreBluetooth                    9.0
pyobjc-framework-CoreData                         9.0
pyobjc-framework-CoreHaptics                      9.0
pyobjc-framework-CoreLocation                     9.0
pyobjc-framework-CoreMedia                        9.0
pyobjc-framework-CoreMediaIO                      9.0
pyobjc-framework-CoreMIDI                         9.0
pyobjc-framework-CoreML                           9.0
pyobjc-framework-CoreMotion                       9.0
pyobjc-framework-CoreServices                     9.0
pyobjc-framework-CoreSpotlight                    9.0
pyobjc-framework-CoreText                         9.0
pyobjc-framework-CoreWLAN                         9.0
pyobjc-framework-CryptoTokenKit                   9.0
pyobjc-framework-DataDetection                    9.0
pyobjc-framework-DeviceCheck                      9.0
pyobjc-framework-DictionaryServices               9.0
pyobjc-framework-DiscRecording                    9.0
pyobjc-framework-DiscRecordingUI                  9.0
pyobjc-framework-DiskArbitration                  9.0
pyobjc-framework-DVDPlayback                      9.0
pyobjc-framework-EventKit                         9.0
pyobjc-framework-ExceptionHandling                9.0
pyobjc-framework-ExecutionPolicy                  9.0
pyobjc-framework-ExternalAccessory                9.0
pyobjc-framework-FileProvider                     9.0
pyobjc-framework-FileProviderUI                   9.0
pyobjc-framework-FinderSync                       9.0
pyobjc-framework-FSEvents                         9.0
pyobjc-framework-GameCenter                       9.0
pyobjc-framework-GameController                   9.0
pyobjc-framework-GameKit                          9.0
pyobjc-framework-GameplayKit                      9.0
pyobjc-framework-ImageCaptureCore                 9.0
pyobjc-framework-IMServicePlugIn                  9.0
pyobjc-framework-InputMethodKit                   9.0
pyobjc-framework-InstallerPlugins                 9.0
pyobjc-framework-InstantMessage                   9.0
pyobjc-framework-Intents                          9.0
pyobjc-framework-IntentsUI                        9.0
pyobjc-framework-IOSurface                        9.0
pyobjc-framework-iTunesLibrary                    9.0
pyobjc-framework-KernelManagement                 9.0
pyobjc-framework-LatentSemanticMapping            9.0
pyobjc-framework-LaunchServices                   9.0
pyobjc-framework-libdispatch                      9.0
pyobjc-framework-LinkPresentation                 9.0
pyobjc-framework-LocalAuthentication              9.0
pyobjc-framework-LocalAuthenticationEmbeddedUI    9.0
pyobjc-framework-MailKit                          9.0
pyobjc-framework-MapKit                           9.0
pyobjc-framework-MediaAccessibility               9.0
pyobjc-framework-MediaLibrary                     9.0
pyobjc-framework-MediaPlayer                      9.0
pyobjc-framework-MediaToolbox                     9.0
pyobjc-framework-Metal                            9.0
pyobjc-framework-MetalKit                         9.0
pyobjc-framework-MetalPerformanceShaders          9.0
pyobjc-framework-MetalPerformanceShadersGraph     9.0
pyobjc-framework-MetricKit                        9.0
pyobjc-framework-MLCompute                        9.0
pyobjc-framework-ModelIO                          9.0
pyobjc-framework-MultipeerConnectivity            9.0
pyobjc-framework-NaturalLanguage                  9.0
pyobjc-framework-NetFS                            9.0
pyobjc-framework-Network                          9.0
pyobjc-framework-NetworkExtension                 9.0
pyobjc-framework-NotificationCenter               9.0
pyobjc-framework-OpenDirectory                    9.0
pyobjc-framework-OSAKit                           9.0
pyobjc-framework-OSLog                            9.0
pyobjc-framework-PassKit                          9.0
pyobjc-framework-PencilKit                        9.0
pyobjc-framework-Photos                           9.0
pyobjc-framework-PhotosUI                         9.0
pyobjc-framework-PreferencePanes                  9.0
pyobjc-framework-PushKit                          9.0
pyobjc-framework-Quartz                           9.0
pyobjc-framework-QuickLookThumbnailing            9.0
pyobjc-framework-ReplayKit                        9.0
pyobjc-framework-SafariServices                   9.0
pyobjc-framework-SceneKit                         9.0
pyobjc-framework-ScreenCaptureKit                 9.0
pyobjc-framework-ScreenSaver                      9.0
pyobjc-framework-ScreenTime                       9.0
pyobjc-framework-ScriptingBridge                  9.0
pyobjc-framework-SearchKit                        9.0
pyobjc-framework-Security                         9.0
pyobjc-framework-SecurityFoundation               9.0
pyobjc-framework-SecurityInterface                9.0
pyobjc-framework-ServiceManagement                9.0
pyobjc-framework-ShazamKit                        9.0
pyobjc-framework-Social                           9.0
pyobjc-framework-SoundAnalysis                    9.0
pyobjc-framework-Speech                           9.0
pyobjc-framework-SpriteKit                        9.0
pyobjc-framework-StoreKit                         9.0
pyobjc-framework-SyncServices                     9.0
pyobjc-framework-SystemConfiguration              9.0
pyobjc-framework-SystemExtensions                 9.0
pyobjc-framework-UniformTypeIdentifiers           9.0
pyobjc-framework-UserNotifications                9.0
pyobjc-framework-UserNotificationsUI              9.0
pyobjc-framework-VideoSubscriberAccount           9.0
pyobjc-framework-VideoToolbox                     9.0
pyobjc-framework-Virtualization                   9.0
pyobjc-framework-Vision                           9.0
pyobjc-framework-WebKit                           9.0
pyparsing                                         3.0.9
PyPDF2                                            2.12.1
PyQt3D                                            5.15.5
PyQt5                                             5.15.7
PyQt5-sip                                         12.11.0
PyQtChart                                         5.15.6
PyQtDataVisualization                             5.15.5
PyQtNetworkAuth                                   5.15.5
PyQtPurchasing                                    5.15.5
PyQtWebEngine                                     5.15.6
python-dateutil                                   2.8.2
pyttsx3                                           2.90
pytz                                              2022.6
PyYAML                                            6.0
requests                                          2.28.1
rsa                                               4.9
s3transfer                                        0.6.0
scipy                                             1.9.3
setproctitle                                      1.3.2
setuptools                                        65.4.1
six                                               1.16.0
SpeechRecognition                                 3.9.0
tempora                                           5.1.0
tinydb                                            4.7.0
tk                                                0.1.0
tqdm                                              4.64.1
types-pytz                                        2022.6.0.1
typing_extensions                                 4.4.0
urllib3                                           1.26.13
Wand                                              0.6.10
wheel                                             0.38.4
zc.lockfile                                       2.0

**Reply by blue-j** [reply]
Python 3.10.8

**Comment by fredsavoir** [comment]
How to you launch the script?

Could you try something = python3 -m myScrip.py

Thanks,

Fred

**Reply by blue-j** [reply]
`python3.10: Error while finding module specification for '/Users/blue-j/Desktop/chromaticity_1976UCS.py'`

if i use the -m switch

**Comment by fredsavoir** [comment]
So :) Just warning so far so good.

**Comment by fredsavoir** [comment]
Don’t add .py in the argument with -m

**Comment by blue-j** [comment]
it suddenly started working fine.  i think it was your good mojo, both of you.  mojo is the primary IT tool!   much gratitude!!   - J

**Comment by fredsavoir** [comment]
Your welcome have fun let’s close the case.
